c10/xpu/XPUStream.cpp

```diff
@@ -37,27 +37,29 @@ thread_local std::unique_ptr<StreamId[]> current_streams = nullptr;
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~
 // How do we assign stream IDs?
 //
-// -- 56 bits --  -- 5 bits -----  -- 3 bits --  -- 1 bits --
-//     zeros      StreamIdIndex    StreamIdType  Ext/native stream
+// -- 55 bits --    -- 5 bits --     -- 3 bits --     -- 1 bit --
+//     zeros       StreamIdIndex     StreamIdType    Ext/native stream
+//                ignored for ext   ignored for ext
 //
 // Where StreamIdType:
 //  000 = normal priority queue
 //  001 = high priority queue
+//  111 = external queue
 //
-// for external stream, StreamID is a sycl::queue* pointer
-// this means that last bit will always be 0
-// so when constructing StreamId for a native stream we set last bit to 1
-// to distinguish between native and external streams
+// For external stream, StreamID is a sycl::queue* pointer. This means that last
+// bit will always be 0. So when constructing StreamId for a native stream we
+// set last bit to 1 to distinguish between native and external streams.
```

Comment:
Pls. highlight the sycl::queue pointer limitation here, as well.

Reply:
Updated.